eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -345,6 +491,119 @@ class Peer(val nodeParams: NodeParams,
       }
       stay()
 
+    case Event(current: CurrentFeerates, d) =>
+      d match {
+        case d: ConnectedData => d.peerConnection ! nodeParams.recommendedFeerates(remoteNodeId, current.feeratesPerKw, d.localFeatures, d.remoteFeatures)
+        case _ => ()
+      }
+      stay()
+
+    case Event(current: CurrentBlockHeight, d) =>
+      // If we have pending will_add_htlc that are timing out, it doesn't make any sense to keep them, even if we have
+      // already funded the corresponding channel: our peer will force-close if we relay them.
+      // Our only option is to fail the upstream HTLCs to ensure that the upstream channels don't force-close.
+      // Note that we won't be paid for the liquidity we've provided, but we don't have a choice.
+      val expired = pendingOnTheFlyFunding.filter {
+        case (_, pending) => pending.proposed.exists(_.htlc.expiry.blockHeight <= current.blockHeight)
+      }
+      expired.foreach {
+        case (paymentHash, pending) =>
+          log.warning("will_add_htlc expired for payment_hash={}, our peer may be malicious", paymentHash)
+          Metrics.OnTheFlyFunding.withTag(Tags.OnTheFlyFundingState, Tags.OnTheFlyFundingStates.Timeout).increment()
+          pending.createFailureCommands().foreach { case (channelId, cmd) => PendingCommandsDb.safeSend(register, nodeParams.db.pendingCommands, channelId, cmd) }
+      }
+      expired.foreach {
+        case (paymentHash, pending) => pending.status match {
+          case _: OnTheFlyFunding.Status.Proposed => ()
+          case _: OnTheFlyFunding.Status.Funded => nodeParams.db.liquidity.removePendingOnTheFlyFunding(remoteNodeId, paymentHash)
+        }
+      }
+      pendingOnTheFlyFunding = pendingOnTheFlyFunding.removedAll(expired.keys)
+      d match {
+        case Peer.Nothing => stay()
+        case d if d.channels.isEmpty && pendingOnTheFlyFunding.isEmpty => stopPeer()
```

Comment:
This is too aggressive for Phoenix: they get constantly disconnected if they don't yet have a channel.

Also, in any case we must explicitly kill the connection when stopping the peer:
```scala
d.peerConnection ! PeerConnection.Kill(KillReason.NoRemainingChannel)
```

Reply:
Yes that's only for the disconnected case, your fix LGTM